modules/backend/src/main/scala/database.doobie.scala

```diff
@@ -1,28 +1,71 @@
-import doobie.*, doobie.implicits.*, doobie.hikari.*
-import cats.effect.IO
+import cats.implicits.*
+import skunk.{Session, Command}
+import skunk.data.Completion
+import fs2.Stream
+import cats.effect.{IO, Resource}
+import natchez.Trace
 
-class DoobieDatabase private (transactor: Transactor[IO]) extends Database:
-  override def stream[I, O](query: SqlOp[I, O]): fs2.Stream[cats.effect.IO, O] =
-    query match
-      case sq: SqlQuery[I, O] => sq.out.stream.transact(transactor)
-      case sq: SqlUpdate[?] => fs2.Stream.eval(sq.out.run.transact(transactor))
-end DoobieDatabase
-
-object DoobieDatabase:
-  def hikari(creds: PgCredentials) =
-    ExecutionContexts
-      .fixedThreadPool[IO](32)
-      .flatMap { ec =>
-        HikariTransactor.newHikariTransactor(
-          "org.postgresql.Driver", // driver classname
-          s"jdbc:postgresql://${creds.host}:${creds.port}/${creds.database}", // connect URL (driver-specific)
-          s"${creds.user}",             // user
-          creds.password.getOrElse(""), // password
-          ec
+class SkunkDatabase private (makeSession: Resource[IO, Session[IO]])
+    extends Database:
+
+  private val defaultChunkSize = 512
+
+  private val transactionalSession = for
+    session     <- makeSession
+    transaction <- session.transaction
+  yield (session, transaction)
+
+  private def transact[A](body: Session[IO] => Stream[IO, A]): Stream[IO, A] =
+    Stream.resource(transactionalSession).flatMap { (session, _) =>
+      body(session)
+    }
+
+  private def transact[A](body: Session[IO] => IO[A]): IO[A] =
+    transactionalSession.use { (session, _) =>
+      body(session)
+    }
+
+  private def dmlAffectedCount(
+      completion: Completion,
+      command: Command[?]
+  ): IO[Int] = completion match
+    case Completion.Update(count) => count.pure[IO]
+    case Completion.Insert(count) => count.pure[IO]
+    case Completion.Delete(count) => count.pure[IO]
+    case x =>
+      IO.raiseError(
+        new RuntimeException(
+          s"Unexpected completion $x, SQL was ${command.sql}"
         )
-      }
-      .map(new DoobieDatabase(_))
-end DoobieDatabase
+      )
+
+  override def stream[I, O](query: SqlOp[I, O]): fs2.Stream[IO, O] =
+    query match
+      case sq: SqlQuery[I, O] =>
+        transact { session =>
+          session.stream(sq.out, sq.input, defaultChunkSize)
+        }
+      case sq: SqlUpdate[?] =>
+        Stream.eval(transact { session =>
+          session
+            .execute(sq.out)(sq.input)
+            .flatMap(completion => dmlAffectedCount(completion, sq.out))
+        })
+end SkunkDatabase
+
+object SkunkDatabase:
+  def make(creds: PgCredentials)(using trace: Trace[IO]) =
+    Session
+      .pooled[IO](
+        host = creds.host,
+        port = creds.port,
+        user = creds.user,
+        password = creds.password,
+        database = creds.database,
+        // TODO double-check if this is equivalent with the hikari setup
+        max = 32
```

Comment:
This is mega-dependent on the project, I think it should be part of configuration. PG serverside max is 100 by default, minus a few superuser connections.